cmd/pebble/cmd_run.go

```diff
@@ -166,6 +168,20 @@ func runDaemon(rcmd *cmdRun, ch chan os.Signal) error {
 
 	logger.Debugf("activation done in %v", time.Now().Truncate(time.Millisecond).Sub(t0))
 
+	if rcmd.Args != nil {
+		mappedArgs, err := convertArgs(rcmd.Args)
+		if err != nil {
+			return err
+		}
+		servopts := client.ServiceOptions{Args: mappedArgs}
+		changeID, err := rcmd.client.PassServiceArgs(&servopts)
```

Comment:
I realize that's how we do it for auto-starting services (below). That is a little different as we want that to be async. But still, I'm not quite sure why we're doing it there either. I've asked @niemeyer on Mattermost [here](https://chat.charmhub.io/charmhub/pl/ofsb9rpn938t8d5qnnr5zw11sa) ... but if we want to change that it can be a separate issue/PR.

Reply:
Took your advice and passing via daemon. Thank you.
Also I don't seem to have the access to the charmhub.io mattermost server. :confused: